examples/typescript-react/js/app.tsx

```diff
@@ -0,0 +1,175 @@
+/*jshint quotmark:false */
+/*jshint white:false */
+/*jshint trailing:false */
+/*jshint newcap:false */
+/*global React, Router*/
+
+/// <reference path="../typings/tsd.d.ts" />
+declare var Router : any;
+
+var TodoModel = app.models.TodoModel;
+var TodoFooter = app.components.TodoFooter;
+var TodoItem = app.components.TodoItem;
+
+namespace app.components {
+
+  export class TodoApp extends React.Component<IAppProps, IAppState> {
+
+    constructor(props : IAppProps) {
+      super(props);
+      this.state = {
+        nowShowing: app.constants.ALL_TODOS,
+        editing: null
+      };
+    }
+
+    public componentDidMount() {
+      var setState = this.setState;
+      var router = Router({
+        '/': setState.bind(this, {nowShowing: app.constants.ALL_TODOS}),
+        '/active': setState.bind(this, {nowShowing: app.constants.ACTIVE_TODOS}),
+        '/completed': setState.bind(this, {nowShowing: app.constants.COMPLETED_TODOS})
+      });
+      router.init('/');
+    }
+
+    public handleNewTodoKeyDown(event) {
+      if (event.keyCode !== app.constants.ENTER_KEY) {
+        return;
+      }
+
+      event.preventDefault();
+
+      var val = React.findDOMNode<HTMLInputElement>(this.refs["newField"]).value.trim();
+
+      if (val) {
+        this.props.model.addTodo(val);
+        React.findDOMNode<HTMLInputElement>(this.refs["newField"]).value = '';
+      }
+    }
+
+    public toggleAll(event) {
```

Comment:
I think it might be better to have explicit types for these instead of `any` implicitly..